src/dodal/devices/ophyd_async_xspress3_mini/xspress3_mini_oa.py

```diff
@@ -0,0 +1,109 @@
+from enum import Enum
+
+from ophyd_async.core import (
+    AsyncStatus,
+    Device,
+    DeviceVector,
+    set_and_wait_for_value,
+    wait_for_value,
+)
+from ophyd_async.epics.signal.signal import (
+    epics_signal_r,
+    epics_signal_rw,
+    epics_signal_rw_rbv,
+)
+
+from dodal.devices.ophyd_async_xspress3_mini.xspress3_mini_channel_oa import (
+    Xspress3MiniChannel,
+)
+
+from dodal.log import LOGGER
+
+
+class AttenuationOptimisationFailedException(Exception):
+    pass
+
+
+class TriggerMode(str, Enum):
+    SOFTWARE = "Software"
+    HARDWARE = "Hardware"
+    BURST = "Burst"
+    TTL_Veto_Only = "TTL_Veto_Only"
+    IDC = "IDC"
+    SOTWARE_START_STOP = "Software_Start/Stop"
+    TTL_BOTH = "TTL_Both"
+    LVDS_VETO_ONLY = "LVDS_Veto_Only"
+    LVDS_both = "LVDS_Both"
+
+
+class UpdateRBV(str, Enum):
+    DISABLED = "Disabled"
+    ENABLED = "Enabled"
+
+
+class EraseState(str, Enum):
+    DONE = 0
```

Comment:
Why str when it's an integer?

Reply:
Should follow the format of `DONE = "DONE"`, ditto for the others

Reply:
The situation is actually worse than I though, what is the best way to deal with undefined Enum, atm I kind of side step it by giving it a enum but set it with int.

```
NotConnected: 
express3_i20: NotConnected:
    erase: TypeError: BL20I-EA-DET-06:ERASE has choices (' ', ' '), which do not match <enum 'EraseState'>, which has ('Done', 'Erasing')
express3_i3: NotConnected:
    erase: TypeError: BL03I-EA-XSP3-01:ERASE has choices (' ', ' '), which do not match <enum 'EraseState'>, which has ('Done', 'Erasing')
```
Also a related issue, what do we do if the rbv return a different enum, would it be reasonable to make 2 signals?
```
class AcquireRBVState(str, Enum):
    DONE = "Done"
    ACQUIRE = "Acquiring"

class AcquireState(str, Enum):
    DONE = "Done"
    ACQUIRE = "Acquire"

  self.acquire = epics_signal_rw(AcquireState, prefix + "Acquire")
  self.acquire_rbv = epics_signal_r(AcquireRBVState, prefix + "Acquire_RBV")
```

Reply:
Yes, you would have to make two signals, see similar issue on the shutter https://github.com/DiamondLightSource/dodal/pull/480#discussion_r1606783981

Reply:
```
NotConnected: 
express3_i20: NotConnected:
    erase: TypeError: BL20I-EA-DET-06:ERASE has choices (' ', ' '), which do not match <enum 'EraseState'>, which has ('Done', 'Erasing')
express3_i3: NotConnected:
    erase: TypeError: BL03I-EA-XSP3-01:ERASE has choices (' ', ' '), which do not match <enum 'EraseState'>, which has ('Done', 'Erasing')

```
Not sure how to handle this,

Below is the only way I can get it to connect and passes all the check/test but I am not sure it will behave correctly with the real device.

```
class EraseState(str, Enum):
    DONE = ""
    ERASE = ""


await self.erase.set(EraseState.ERASE)  
```

`(int, enum)`  should work but backend would not allow it to connect same for just int type.

Anyone has an idea what's the best way to deal with this?

Reply:
Just an up date below will work, dont know if it is intentional.
```
class EraseState(str, Enum):
    DONE = ""
    ERASE = ""


await self.erase.set(<int>)  int can be the position of the c enum  
```

Reply:
Tagging @stan-dot as this is general good info. So to look up the type of a PV I do the following:
* `python2 /dls_sw/work/useful-scripts/find-pv.py BL20I-EA-DET-06:ERASE`
* This showed me the PV was defined in `/dls_sw/work/R3.14.12.7/support/BL20I-BUILDER/iocs/BL20I-EA-IOC-05/db/BL20I-EA-IOC-05_expanded.db` as a `bo` record.
* EPICS record types are documented at https://epics.anl.gov/base/R3-15/7-docs/RecordReference.html
* If you look at https://epics.anl.gov/base/R3-15/7-docs/boRecord.html you can see that `bo` records take values of 0 or 1, but they can define `ZNAM` and `ONAM` which are user friendly names for the 0/1 states
* So if we do:
```
$ caget BL03I-EA-XSP3-01:ERASE.ZNAM BL03I-EA-XSP3-01:ERASE.ONAM
BL03I-EA-XSP3-01:ERASE.ZNAM    
BL03I-EA-XSP3-01:ERASE.ONAM    
```
We can see that both are blank compared to a PV of the same type I found elsewhere:
```
$ caget BL03I-MO-SGON-01:CS1:DeferMoves.ZNAM BL03I-MO-SGON-01:CS1:DeferMoves.ONAM
BL03I-MO-SGON-01:CS1:DeferMoves.ZNAM Defer Off
BL03I-MO-SGON-01:CS1:DeferMoves.ONAM Defer On
```
* Given they're both `blank` this is why your string enum complains if they're anything else. Obviously this won't actually work in production as if you set `blank` on the PV how does it know which one you actually need?

So, we have some choices:
1. Talk to controls and ask them to provide named states for the PV, or change the PV to a type that is more appropriate
2. Use an `int` - This will work but is a bit of a shame as it's not super clear that writing 1 to the erase is what you want to do
3. Use an `IntEnum`  e.g.
```python
class EraseEnum(int, Enum):
    ERASE = 1
    DONE = 0
```
But are you saying that didn't work? In which case we should probably open a discussion up about if this is a bug in `ophyd_async`

I prefer option 1 but it will be the most work. I can live with option 2 for now in the interests of just getting things done, maybe we can then put in a ticket with controls to improve it at a later date

Reply:
Thanks, knowing how to check pv type would have save me so much time :).

Both 2 and 3 will fail to connect when opyhd_asynco beckend type check the signal as it expect str type enum. As it is the only way for it to work is to trick opyhd_asynco  with two blanks for connection and set it with int later, which is not the best idea.
Who would be a good person to talk to about epics xspress3?  

> * Given they're both `blank` this is why your string enum complains if they're anything else. Obviously this won't actually work in production as if you set `blank` on the PV how does it know which one you actually need?

Something I notice playing around with the epic pv I have access to, if I set the Names  to "blank" they default into int, and option 2 works.  But it seem not all blanks are equal.

Reply:
Ok, I think that is a bug in `ophyd_async` too then as we can't assume that all PVs will have valid string representations. Could you make an issue in `ophyd_async` about it?

> Who would be a good person to talk to about epics xspress3?
I will find out for you

Reply:
Just for record keeping, issue raised in [ophyd_async] (https://github.com/bluesky/ophyd-async/issues/353).

After discussion with control, the problem signal is being use in any form so it is removed from this device and it is no longer an issue for this pr.